internal/provider/provider_framework.go

```diff
@@ -0,0 +1,194 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-resource-manager/stable/2019-12-10/client/organization_service"
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-resource-manager/stable/2019-12-10/client/project_service"
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-resource-manager/stable/2019-12-10/models"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	diagnostic "github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/provider"
+	"github.com/hashicorp/terraform-plugin-framework/provider/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+// This is an implementation using the Provider framework
+type ProviderFramework struct {
+	version string
+}
+
+type ProviderFrameworkConfiguration struct {
+	Client *clients.Client
+}
+
+type ProviderFrameworkModel struct {
+	ClientSecret types.String `tfsdk:"client_secret"`
+	ClientID     types.String `tfsdk:"client_id"`
+	ProjectID    types.String `tfsdk:"project_id"`
+}
+
+func (p *ProviderFramework) Metadata(ctx context.Context, req provider.MetadataRequest, resp *provider.MetadataResponse) {
+	resp.TypeName = "hcp"
+	resp.Version = "dev"
+}
+
+func (p *ProviderFramework) Schema(ctx context.Context, req provider.SchemaRequest, resp *provider.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"client_id": schema.StringAttribute{
+				Optional:    true,
+				Description: "The OAuth2 Client ID for API operations.",
+			},
+			"client_secret": schema.StringAttribute{
+				Optional:    true,
+				Description: "The OAuth2 Client Secret for API operations.",
+			},
+			"project_id": schema.StringAttribute{
+				Optional:    true,
+				Description: "The default project in which resources should be created.",
+			},
+		},
+	}
+}
+
+// TODO add check for HCP env vars
```

Comment:
Is this TODO still valid?

Reply:
whooops. Lemme double check.

Reply:
Definitely forgot to add this. Thanks for catching